classes/class-p4-campaign-importer.php

```diff
@@ -281,5 +282,26 @@ public function process_campaign_metas( $post_meta ) {
 
 			return $post_meta;
 		}
+
+		/**
+		 * Skip already existing postmeta data from import.
+		 *
+		 * @param array   $postmeta The to be imported post meta fields.
+		 * @param integer $post_id Post ID.
+		 *
+		 * @return array The cleaned post meta fields.
+		 */
+		public function skip_duplicate_postmeta_import( $postmeta, $post_id ) {
+			$existing_postmeta = get_post_meta( $post_id );
+			if ( $postmeta && $existing_postmeta ) {
```

Comment:
You can use `array_filter` on `$postmeta` with function returning `!array_key_exists( $meta['key'], $existing_postmeta )`.

Reply:
Hi @Inwerpsel , The callback function of `array_filter` won't get access to `$existing_postmeta` var. I tried with with `global` declaration, but it's not working.